be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -379,11 +381,14 @@ StatusOr<RowsetSharedPtr> HorizontalBetaRowsetWriter::build() {
 // when: segment files number larger than one, no delete files(for now, ignore it when just one segment)
 // how: for final merge scenario, temporary files created at first, merge them, create final segment files.
 Status HorizontalBetaRowsetWriter::_final_merge() {
-    if (_num_segment == 1) {
-        RETURN_IF_ERROR_WITH_WARN(
-                _fs->rename_file(Rowset::segment_temp_file_path(_context.rowset_path_prefix, _context.rowset_id, 0),
-                                 Rowset::segment_file_path(_context.rowset_path_prefix, _context.rowset_id, 0)),
-                "Fail to rename file");
+    if (_num_segment == 1 || !_context.schema_change_sorting) {
+        for (size_t seg_id = 0; seg_id < _num_segment; seg_id++) {
+            RETURN_IF_ERROR_WITH_WARN(
+                    _fs->rename_file(
+                            Rowset::segment_temp_file_path(_context.rowset_path_prefix, _context.rowset_id, seg_id),
```

Comment:
No need to rename file

Reply:
> No need to rename file

done, changed in https://github.com/StarRocks/starrocks/pull/10903/commits/04a55a7ee3f427c68ea05b8f7cfaa7c092b3dd2d